go/pools/resource_pool.go

```diff
@@ -37,6 +37,8 @@ var (
 
 	// ErrTimeout is returned if a resource get times out.
 	ErrTimeout = errors.New("resource pool timed out")
+
+	prefillTimeout = 30 * time.Second
```

Comment:
May not be enough based on our testing. On a cold start opening 600 connections takes about ~38s.

Reply:
So maybe make this configurable?

Reply:
:man_shrugging: Yet another flag. I'll begrudgingly add it.

Reply:
I know this is getting to be a lot, but I just don't see how we can make a one-size-fits-all solution here.

Reply:
I'd say let's go with 30 seconds for now. If it actually becomes a problem in real life then we can address it by making it configurable then.

Reply:
:tada: